api/src/lib/openapi/fetch.ts

```diff
@@ -0,0 +1,60 @@
+import type { LibSQLDatabase } from "drizzle-orm/libsql";
+import type * as schema from "../../db/schema.js";
+import logger from "../../logger/index.js";
+import { resolveServiceArg } from "../../probe-routes.js";
+import { getAllSettings } from "../settings/index.js";
+import type { OpenApiSpec } from "./types.js";
+
+export async function fetchOpenApiSpec(db: LibSQLDatabase<typeof schema>) {
+  const specUrl = await getSpecUrl(db);
+  if (!specUrl) {
+    logger.debug("No OpenAPI spec URL found");
+    return null;
+  }
+  const resolvedSpecUrl = resolveSpecUrl(specUrl);
+  if (!resolvedSpecUrl) {
+    logger.debug("No resolved OpenAPI spec URL found");
+    return null;
+  }
+  logger.debug(`Fetching OpenAPI spec from ${resolvedSpecUrl}`);
+  const response = await fetch(resolvedSpecUrl, {
+    headers: {
+      // NOTE - This is to avoid infinite loops when the OpenAPI spec is fetched from Studio
+      //        We need to make sure that the user has instrumented their app with @fiberplane/hono-otel >= 0.4.1
```

Comment:
this part is really crucial - we need to validate that the user has the correct version of hono-otel